sui/its.js

```diff
@@ -365,35 +386,27 @@ async function migrateCoinMetadata(keypair, client, config, contracts, args, opt
 }
 
 // give_unlinked_coin
-async function giveUnlinkedCoin(keypair, client, config, contracts, args, options) {
+async function giveUnlinkedCoin(keypair, client, _, contracts, args, options) {
     const { InterchainTokenService: itsConfig, AxelarGateway } = contracts;
     const { InterchainTokenService } = itsConfig.objects;
     const walletAddress = keypair.toSuiAddress();
-    const deployConfig = { client, keypair, options, walletAddress };
-    const [symbol, name, decimals] = args;
+    const [symbol, tokenId] = args;
     const txBuilder = new TxBuilder(client);
 
-    if (options.salt) {
-        validateParameters({
-            isHexString: { salt: options.salt },
-        });
-    }
+    validateParameters({
+        isHexString: { tokenId },
+    });
 
-    // Deploy token on Sui
-    const [metadata, packageId, tokenType, treasuryCap] = await deployTokenFromInfo(deployConfig, symbol, name, decimals);
+    const coin = contracts[symbol.toUpperCase()];
+    if (!coin) {
+        throw new Error(`Cannot find coin with symbol ${symbol} in config`);
+    }
 
-    // Register deployed token (custom)
-    const [tokenId, _channelId, saltAddress, _result] = await registerCustomCoinUtil(
-        deployConfig,
-        itsConfig,
-        AxelarGateway,
-        symbol,
-        metadata,
-        tokenType,
-        null,
-        options.salt ? options.salt : null,
-    );
-    if (!tokenId) throw new Error(`error resolving token id from registration tx, got ${tokenId}`);
+    const decimals = coin.decimals;
+    const metadata = coin.objects.Metadata;
+    const packageId = coin.address;
+    const tokenType = coin.typeArgument;
+    const treasuryCap = coin.objects.TreasuryCap;
```

Comment:
as part of removing the need to store token data in the config, this should be refactored to take in the coin type, and dynamically query any metadata needed

Reply:
this work is partially completed in another PR, and removing tokens from config is a task in JIRA